### PR TITLE
Use &’static str instead of c_char*

### PR DESCRIPTION
**Pull request by luyikk**
This is safer and error impl send is easier to use.

### CODE REVIEW

**Comment by Michael-F-Bryan** [comment]
Thanks for making this PR @luyikk! Is there anything else you need to do before it can be merged? 

I'd like to use this change because it makes `Wasm3Error` implement `Send` and `Sync`.

File: Cargo.toml [hunk 1]
```diff
@@ -33,6 +34,7 @@ package = "wasm3-sys"
 
 [dev-dependencies]
 trybuild = "1.0"
+anyhow = "1"
```
Comment (Veykril) [inline Cargo.toml:37]: As stated in my last PR, this should not need to touch the examples or add this dependency. The examples are fine as is, the `expects` are intentional here.

File: src/error.rs [hunk 2]
```diff
@@ -4,6 +4,35 @@ use core::fmt;
...
+/// Stack overflow error static str
+pub static ref STACK_OVERFLOW_STR:&'static str=unsafe { cstr_to_str(ffi::m3Err_trapStackOverflow) };
+
+/// function lookup failed error static str
+pub static ref FUNCTION_LOOKUP_FAILED_STR:&'static str=unsafe { cstr_to_str(ffi::m3Err_functionLookupFailed) };
+/// malloc failed error static str
+pub static ref MALLOC_FAILED_STR:&'static str=unsafe { cstr_to_str(ffi::m3Err_mallocFailed) };
+
+}
+
```
Comment (Veykril) [inline src/error.rs:35]: I'd prefer not to introduce `lazy_static`, ideally `cstr_to_str` could be made `const`, but the std lib functions for that aren't const stable yet.

So I think we should do the conversion in the constructor of the error type.

File: src/error.rs [hunk 3]
```diff
@@ -118,14 +151,14 @@ impl Error {
         if ptr.is_null() {
             Ok(())
         } else if unsafe { ptr == ffi::m3Err_functionLookupFailed } {
-            Err(Error::FunctionNotFound)
```
Comment (Veykril) [inline src/error.rs:121]: `FunctionNotFound` is unused now right? In that case we can remove that variant.

**Review by Veykril — COMMENTED** [review]
This fell off my radar hence the late review sorry

**Pull request closed without merging**